### PR TITLE
setState only if the component is still mounted

### PR DESCRIPTION
**Pull request by Crash--**

### CODE REVIEW

**Comment by henrikra** [comment]
Does this increase performance or what?

**Comment by Crash--** [comment]
Don't think so. It just because sometime my component is unmounted before the end of the async fetch

File: index.js [hunk 1]
```diff
@@ -66,24 +66,33 @@ class SvgUri extends Component{
     this.obtainComponentAtts  = this.obtainComponentAtts.bind(this);
     this.inspectNode          = this.inspectNode.bind(this);
     this.fecthSVGData         = this.fecthSVGData.bind(this);
-
+    let responseXML;
```
Comment (matc4) [inline index.js]: this.responseXML;

File: index.js [hunk 2]
```diff
@@ -66,24 +66,33 @@ class SvgUri extends Component{
     this.obtainComponentAtts  = this.obtainComponentAtts.bind(this);
     this.inspectNode          = this.inspectNode.bind(this);
     this.fecthSVGData         = this.fecthSVGData.bind(this);
-
+    let responseXML; 
+    let isComponentMounted = false;
```
Comment (matc4) [inline index.js]: this.isComponentMounted = false;

File: index.js [hunk 3]
```diff
@@ -66,24 +66,33 @@ class SvgUri extends Component{
...
+      this.isComponentMounted = false
+    }
   async fecthSVGData(uri){
      try {
-         let response = await fetch(uri);
-         let responseXML = await response.text();
-         this.setState({svgXmlData:responseXML});
-         return responseXML;
+        let response = await fetch(uri);
+        let responseXML = await response.text();
```
Comment (matc4) [inline index.js]: this.responseXML = await response.text();

File: index.js [hunk 4]
```diff
@@ -66,24 +66,33 @@ class SvgUri extends Component{
...
+        this.responseXML = responseXML
```
Comment (matc4) [inline index.js]: No need for this line with the modification in the previous

**Comment by matc4** [comment]
Hi @Crash-- 

Can you make the changes in the constructor?,


Thanks for your help!

**Comment by ruifreitas** [comment]
Hi @Crash--, thank you for your contribution :).

Can you please update your PR with the requested changes and also with a rebase?

Thank you!

**Comment by matc4** [comment]
Added in the last version!

Thanks!

**Pull request closed without merging**